### ISSUE TITLE
Implement as a plugin.

### ISSUE THREAD

**Issue by fennecdjay** [projectM-visualizer/projectm#6]
I'd like to integrate ProjectM to [this](https://fennecdjay.github.io/Gwion), but I do not manage do find the informations I need either on the doc or the code.
While I'm conscious I probably missed something huge, I prefer asking now if anything similar as been done, or if you could consider making a tutorial to this (to which I'd gladly contribute if I manage to get started).
Please tell me if this seems unclear.
Thanks for the beautiful project.

**Comment by revmischa** [comment]
Hello, I’m not super sure what that project is but okay. I don’t think there is much in the way of documentation but I’d be happy to help put some together if you want to take the lead on that.
There are a few simple examples of how to use the library. I was recently making a bare-bones SDL version that I think is a good starting place. See https://github.com/projectM-visualizer/projectm/blob/master/src/projectM-sdl/projectM_SDL_main.cpp

Basically build libprojectm and link it to something like that. You initialize openGL, call `app.pm = new projectM(app.settings);`, then call `app->pm->pcm()->addPCM16(pcm_data);` then `app->pm->renderFrame();` in your mainloop.

**Comment by fennecdjay** [comment]
Thanks for your crystal clear answer, I'm gonna give it a try, and keep notes so we can start bulding a tuto (but I must admit making papers is **NOT** something I'm good at, but I'll try to make it change).

----------

The project is a *music composition/sound synthesis* language, I hope this is somehow clearer.

**Comment by fennecdjay** [comment]
I got it working! thank you!
Yet it needs further refining. when This is done, I'll see what I can do as a tuto.
